src/coreComponents/constitutive/fluid/BlackOilFluidBase.hpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file BlackOilFluidBase.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_FLUID_BLACKOILFLUIDBASE_HPP_
+#define GEOSX_CONSTITUTIVE_FLUID_BLACKOILFLUIDBASE_HPP_
+
+#include "constitutive/fluid/MultiFluidBase.hpp"
+#include "functions/TableFunction.hpp"
+
+
+namespace geosx
+{
+
+namespace constitutive
+{
+
+class BlackOilFluidBase : public MultiFluidBase
+{
+public:
+
+  using exec_policy = parallelDevicePolicy<>;
+
+  struct PhaseType
+  {
+    static constexpr integer OIL            = 0;
+    static constexpr integer GAS            = 1;
+    static constexpr integer WATER          = 2;
+    static constexpr integer MAX_NUM_PHASES = 3;
+  };
+
+  BlackOilFluidBase( string const & name, Group * const parent );
+
+  virtual ~BlackOilFluidBase() override = default;
+
+  struct viewKeyStruct : MultiFluidBase::viewKeyStruct
+  {
+    static constexpr char const * surfacePhaseMassDensitiesString() { return "surfaceDensities"; }
+    static constexpr char const * tableFilesString() { return "tableFiles"; }
+    static constexpr char const * formationVolumeFactorTableNamesString() { return "hydrocarbonFormationVolFactorTableNames"; }
+    static constexpr char const * viscosityTableNamesString() { return "hydrocarbonViscosityTableNames"; }
+    static constexpr char const * waterRefPressureString() { return "waterReferencePressure"; }
+    static constexpr char const * waterFormationVolumeFactorString() { return "waterFormationVolumeFactor"; }
+    static constexpr char const * waterCompressibilityString() { return "waterCompressibility"; }
+    static constexpr char const * waterViscosityString() { return "waterViscosity"; }
+
+  };
```

Comment:
While I respect your "_personal preference_", it's always interesting to get _facts_ in order to weigh your choices instead.

Let's consider the case of `surfacePhaseMassDensitiesString`.
* How do I know the type of the `surfacePhaseMassDensities`?
* Does it make sense to allow different types for `surfacePhaseMassDensities`?
* How do I prevent a developer to register some `surfacePhaseMassDensities` with different and/or inconsistent types?
* If we happen to change the main (unique?) type of `surfacePhaseMassDensities`, it will surely not break at compile time but at runtime with failed casts.